From 789161b82de792ce45f46b8651984318766eff76 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Wed, 8 May 2019 20:02:49 +0200
Subject: [PATCH 204/345] MIPS: Make setting of MAX_PHYSMEM_BITS settable per
 sub-architecture.

OCTEON needs 40 bits (41 for NUMA).  But making this too big
universally wastes space in the mem_section array, so make it settable
per mips machine type.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Carlos Munoz <cmunoz@caviumnetworks.com>
---
 arch/mips/include/asm/mach-cavium-octeon/physmem.h | 10 ++++++++++
 arch/mips/include/asm/mach-generic/physmem.h       |  6 ++++++
 arch/mips/include/asm/sparsemem.h                  |  3 ++-
 3 files changed, 18 insertions(+), 1 deletion(-)
 create mode 100644 arch/mips/include/asm/mach-cavium-octeon/physmem.h
 create mode 100644 arch/mips/include/asm/mach-generic/physmem.h

diff --git a/arch/mips/include/asm/mach-cavium-octeon/physmem.h b/arch/mips/include/asm/mach-cavium-octeon/physmem.h
new file mode 100644
index 000000000000..d1622a94b52d
--- /dev/null
+++ b/arch/mips/include/asm/mach-cavium-octeon/physmem.h
@@ -0,0 +1,10 @@
+#ifndef _ASM_MACH_CAVIUM_OCTEON_PHYSMEM_H
+#define _ASM_MACH_CAVIUM_OCTEON_PHYSMEM_H
+
+#ifdef CONFIG_NUMA
+#define MAX_PHYSMEM_BITS        41
+#else
+#define MAX_PHYSMEM_BITS        40
+#endif
+
+#endif /* _ASM_MACH_CAVIUM_OCTEON_PHYSMEM_H */
diff --git a/arch/mips/include/asm/mach-generic/physmem.h b/arch/mips/include/asm/mach-generic/physmem.h
new file mode 100644
index 000000000000..6a900cd5e188
--- /dev/null
+++ b/arch/mips/include/asm/mach-generic/physmem.h
@@ -0,0 +1,6 @@
+#ifndef _ASM_MACH_GENERIC_PHYSMEM_H
+#define _ASM_MACH_GENERIC_PHYSMEM_H
+
+#define MAX_PHYSMEM_BITS        38
+
+#endif /*_ASM_MACH_GENERIC_PHYSMEM_H */
diff --git a/arch/mips/include/asm/sparsemem.h b/arch/mips/include/asm/sparsemem.h
index b0686ca3d8bf..33f871939b35 100644
--- a/arch/mips/include/asm/sparsemem.h
+++ b/arch/mips/include/asm/sparsemem.h
@@ -12,7 +12,8 @@
 #else
 # define SECTION_SIZE_BITS	28
 #endif
-#define MAX_PHYSMEM_BITS	48
+
+#include <physmem.h>
 
 #endif /* CONFIG_SPARSEMEM */
 #endif /* _MIPS_SPARSEMEM_H */
-- 
2.25.1

